cmd/forwarder.go

```diff
@@ -120,6 +119,20 @@ func Forwarder() *cobra.Command {
 					})
 				}
 			}()
+
+			// listen for signals so can gracefully stop kakfa producer
+			go func() {
+				for _ = range signals {
+					// tell forwarder to stop handling in incoming requests
+					fwd.Stop()
+					time.Sleep(500 * time.Millisecond)
```

Comment:
I don't like `time.Sleep` usage here. `fwd.Stop()` should block and return only once it has stopped, or there should be a channel to listen on from the forwarder that signals it is finished running.

Same with the writer.